### PR TITLE
diffusion: support Qwen-Image-Layered

### PR DESCRIPTION
**Pull request by chhnb**
## Motivation
Co-authored-by: @yhyang201

## Modifications
This PR introduces Qwen-Image-Layered model support in SGLang and enables layered image generation via the `num-frames` argument. Users can download an input image and run
`sglang generate --model-path Qwen/Qwen-Image-Layered --image-path "/sgl-workspace/4.png" --num-frames 4`
to generate multi-layer image outputs from a single input image, with `num-frames` directly mapping to the diffusion layer count.
Input image:

input

Output images:

image
image
image
image

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/models/dits/qwen_image.py [hunk 1]
```diff
@@ -232,6 +244,200 @@ def _compute_video_freqs(
...
+                max_vid_index = max(height, width, max_vid_index)
+
+        max_vid_index = max(max_vid_index, layer_num)
+        max_len = max(txt_seq_lens)
+        txt_freqs = self.pos_freqs[max_vid_index : max_vid_index + max_len, ...]
+        vid_freqs = torch.cat(vid_freqs, dim=0)
+
+        return vid_freqs, txt_freqs
+
+    @functools.lru_cache(maxsize=None)
```
[the remark this answers is not in the document]
Comment (chhnb) [reply python/sglang/multimodal_gen/runtime/models/dits/qwen_image.py:355]: https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/transformers/transformer_qwenimage.py#L344-L346

**Review by yhyang201 — APPROVED** [review]

**Comment by yhyang201** [comment]
Here are a few points to note:

1. This PR enables `sglang generate` to produce multiple image outputs.
2. It incorporates changes inspired by [https://github.com/huggingface/diffusers/pull/12853](https://github.com/huggingface/diffusers/pull/12853), specifically modifying `python/sglang/multimodal_gen/runtime/models/vaes/autoencoder_kl_qwenimage.py`.
3. A new `QwenImageLayeredBeforeDenoisingStage` has been added in `python/sglang/multimodal_gen/runtime/models/model_stages/qwen_image_layered.py`, which is responsible for handling all `QwenImageLayered`–related operations prior to the denoising stage.

cc @mickqian
